frontend/lib/src/hostComm/HostCommunicationManager.tsx

```diff
@@ -14,13 +14,13 @@
  * limitations under the License.
  */
 
+import { AppConfig } from "@streamlit/connection"
```

Comment:
If I have just looked correctly, this seems to be used for typing only. If this and `LibConfig` (see comment [here](https://github.com/streamlit/streamlit/pull/10309/files#r1939907083)) are the only two dependencies on the new `connection` package, I believe we could duplicate the typing for now for usage here (with a comment). In a follow-up, we should probably have a typing package as well

Reply:
To be fair, I feel the config should be in connection because our host config provides all of the information needed, so that typing makes sense. I might argue here that the HostCommunication should be a separate package which depends on the connection package.

The LibConfig is a bit tough because it's used in the View.

Since it is just the types, I wonder if we can leverage devDependencies here. I'll try that out. With tree-shaking and dependencies, it should not be included in the final build.